source/fundamentals/connection.txt

```diff
@@ -4,16 +4,203 @@ Connection Guide
+
+.. figure:: /includes/figures/connection_string_parts.png
+   :alt: Each part of the connection string
+
+In this example, we use ``mongodb`` for the protocol, which specifies the
+:manual:`Standard Connection String Format
+</reference/connection-string/#std-label-connections-standard-connection-string-format>`.
+You can also use the :manual:`DNS Seed List Connection Format
+</reference/connection-string/#dns-seed-list-connection-format>` if you
+want more flexibility of deployment and the ability to change the
+servers in rotation without reconfiguring clients.
+
+.. note::
+
+   If your deployment is on MongoDB Atlas, see the
+   :atlas:`Atlas driver connection guide </driver-connection>`
+   and select Go from the language dropdown to retrieve your connection string.
+
+The next part of the connection string contains your username and, if
+you are using password-based authentication, your password. Replace the value of
+``user`` with your username and ``pass`` with your password. If you are using an
+authentication mechanism that does not require a username and password, omit
+this part of the connection URI.
+
+The next part of the connection string specifies the hostname or IP address and
+port of your MongoDB instance. In the preceding example, we use ``sample.host``
+as the hostname and ``27017`` as the port. Replace these values to point to
+your MongoDB instance.
+
+The last part of the connection string specifies connection and authentication
+options. In the example, we set two connection options:
+``maxPoolSize=20`` and ``w=majority``. For more information on connection
+options, read the :ref:`connection-options` section of this guide.
+
+The following code shows how you can use the sample connection URI in a client to
+connect to MongoDB.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/srv.go
+   :language: go
+
+.. note::
+
+   For information about connecting to Atlas Serverless, see the
+   `Serverless Instance Limitations page
+   <https://docs.atlas.mongodb.com/reference/serverless-instance-limitations/#minimum-driver-versions-for-serverless-instances>`__
+   to identify the minimum driver version you need.
+
+Connect to a MongoDB Server on Your Local Machine
++++++++++++++++++++++++++++++++++++++++++++++++++
+
+.. include:: /includes/localhost-connection.rst
+
+To test whether you can connect to your server, replace the connection
+string with your localhost connection string in the preceding code example.
+
+------------------------
+Connect to a Replica Set
+------------------------
+
+A MongoDB replica set deployment is a group of connected instances that
+store the same set of data. This configuration provides data
+redundancy and high data availability.
+
+To connect to a replica set deployment, specify the hostname and port numbers
+of each instance, separated by commas, and the replica set name as the value
+of the ``replicaSet`` parameter in the connection string. In the following
```

Comment:
Optional: Since the `replicaSet` option is referenced, should this be included in the table of "Connection Options"?

Reply:
Yes, will include.